src/DB.ts

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+import * as mongoose from 'mongoose';
+import * as Promise from 'bluebird';
+const label = '[fh-mongoose] ';
+
+const handlers = {
+  onError: function(err: Error) {
+    console.error(label, err.toString());
+  },
+  onConnection: function(uri: string) {
+    console.info(label, 'Connected to Mongo @', uri);
+  },
+  onConnectionOpen: function() {
+    console.info(label, 'Mongo connection open');
+  },
+  onClose: function() {
+    console.info(label, 'Mongo connection closed');
+  }
+};
+
+export default class DB {
+  connection: mongoose.Connection;
+  connectToMongo = function(this: DB, uri: string, opts: mongoose.ConnectionOptions = {}) {
+    return new Promise<mongoose.Connection>((resolve, reject) => {
```

Comment:
Ideally typescript would pick the generic type correctly from `resolve(connection)`, but since not all paths resolve with the same type we need to declare it specifically.

This is actually an interesting edge case, I'll add it to a 'pitfalls' section in the document.